### DISCUSSION TITLE
PreSync hook enforcement before submit workflow via ui?

### DISCUSSION THREAD

**Discussion by tooptoop4** [argoproj/argo-workflows#8039] [Q&A]
is there a way to enforce PreSync (ie https://argo-cd.readthedocs.io/en/stable/user-guide/resource_hooks/) hooks before submitting any workflow via ui?

ie some argument for RemedyIncidentID then the hook does api call to Remedy to check the INC state and if not in proper state then prevent the workflow from being submitted. If the workflow being submitted does not supply RemedyIncidentID at all then it should also prevent

**Comment by alexec** [comment]
Short answer, no. Future plugins might be able to.
